sky/execution.py

```diff
@@ -460,6 +460,19 @@ def exec(  # pylint: disable=redefined-builtin
     backend_utils.check_cluster_name_not_reserved(cluster_name,
                                                   operation_str='sky.exec')
 
+    if isinstance(entrypoint, sky.Dag):
+        task = entrypoint.tasks[0]
+    else:
+        task = entrypoint
+    task_resources = task.get_resources()
+    if task_resources:
```

Comment:
Is this testing for None? Looks like we always set https://github.com/skypilot-org/skypilot/blob/9324e524d04a3cb3d4de136467bedf5f35334996/sky/task.py#L156.

Reply:
Good point. I thought `resources` can be an empty set, but it seems such a case cannot exist. Removed `if task_resources`.